lib/devise/models/lockable.rb

```diff
@@ -181,7 +181,9 @@ def unlock_access_by_token(unlock_token)
 
         # Is the unlock enabled for the given unlock strategy?
         def unlock_strategy_enabled?(strategy)
-          [:both, strategy].include?(self.unlock_strategy)
+          self.unlock_strategy == strategy ||
+            # only :time and :email are subsets of the :both strategy
+            (self.unlock_strategy == :both && [:time, :email].include?(strategy))
```

Comment:
can you please move the array to a `SUPPORTED_UNLOCK_STRATEGIES` (or something like that) constant?